File: python/cow_bull.py
from random import randint



def play_cow_game(hidden_word):
    count = 0
    while True:
        user_word = input("Enter your number: ")
        #count = is_cow_bull(user_word,hidden_word)
        cow_count,bull_count = is_cow_bull(user_word,hidden_word)
        count += 1
        if cow_count == 4:
            print(f"You won. {count} attempts were taken.")
            break
        else:
            print(f"{cow_count} cow, {bull_count} bull")





def is_cow_bull(user_word,hidden_word):
    cow_count = 0
    bull_count = 0
    for i, c in enumerate(user_word):
        if c == hidden_word[i]:
            cow_count += 1
        elif c in hidden_word:
            bull_count += 1
    return (cow_count,bull_count)


if __name__ == "__main__":
    hidden_word = str(randint(1000, 9999))
    play_cow_game(hidden_word)

File: web/html/rough-work/cow_bull.js
var playing = true;

function isCowBull(userWord, hiddenWord) {
	var cowCount = 0;
	var bullCount = 0;
	for(var i = 0; i < userWord.length; i++){
		if(userWord[i] === hiddenWord[i]) {
			cowCount += 1;
		} else if(hiddenWord.includes(userWord[i])){
			bullCount += 1;
		}
	}
	return [cowCount, bullCount];
}

function playCowGame(hiddenWord) {
	var retryCount = 0;
	while(playing){
		var userWord = $("input").val();
		var counts = isCowBull(userWord, hiddenWord);
		var cowCount = counts[0];
		var bullCount = counts[1];

		retryCount += 1;
		if(cowCount === 4) {
			console.log("You won." + " " + retryCount + " attempts were taken.");
			playing = false;
		} else {
			console.log(cowCount + ": cow" + ", " + bullCount + ": bull");
		}
	}
}


playCowGame("1035");
